url78f6619d18_IntersectAustralia_rifcs_tgz-pJ8-org_ands_rifcs_example_SampleRIFCSBuildCollectionJ8/src/RelatedObject.java

```diff
@@ -0,0 +1,95 @@
+/*     */ package org.ands.rifcs.base;
+/*     */ 
+/*     */ import java.util.ArrayList;
+/*     */ import java.util.List;
+/*     */ import org.w3c.dom.Element;
+/*     */ import org.w3c.dom.Node;
+/*     */ import org.w3c.dom.NodeList;
+/*     */ 
+/*     */ public class RelatedObject extends RIFCSElement
+/*     */ {
+/*  36 */   List<Relation> relations = new ArrayList();
+/*     */ 
+/*     */   protected RelatedObject(Node n)
+/*     */     throws RIFCSException
+/*     */   {
+/*  48 */     super(n, "relatedObject");
+/*  49 */     initStructures();
+/*     */   }
+/*     */ 
+/*     */   public void setKey(String keyValue)
+/*     */   {
+/*  61 */     Element key = newElement("key");
+/*  62 */     key.setTextContent(keyValue);
+/*  63 */     getElement().appendChild(key);
+/*     */   }
+/*     */ 
+/*     */   public String getKey()
+/*     */   {
+/*  75 */     NodeList nl = super.getElements("key");
+/*  76 */     if (nl.getLength() == 1)
+/*     */     {
+/*  78 */       return nl.item(0).getTextContent();
+/*     */     }
+/*     */ 
+/*  81 */     return null;
+/*     */   }
+/*     */ 
+/*     */   public Relation newRelation()
+/*     */     throws RIFCSException
+/*     */   {
+/*  99 */     return new Relation(newElement("relation"));
+/*     */   }
+/*     */ 
+/*     */   public void addRelation(String type, String url, String description, String descriptionLanguage)
+/*     */     throws RIFCSException
+/*     */   {
+/* 120 */     Relation relation = newRelation();
+/*     */ 
+/* 122 */     relation.setType(type);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;type&#39; (index: 0 | type: String). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;description&#39;, &#39;url&#39;, &#39;descriptionLanguage&#39;. The same parameter type is validated here for example: url78f6619d18_IntersectAustralia_rifcs_tgz-pJ8-org_ands_rifcs_example_SampleRIFCSBuildCollectionJ8/src/DateElement.java:16. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).